packages/components/tests/integration/components/hds/alert/index-test.js

```diff
@@ -0,0 +1,214 @@
+    );
+    assert.dom(this.element).hasText('This is the description');
+  });
+  test('it should render both the title and the description when both the @title and @description arguments are provided', async function (assert) {
+    assert.expect(2);
+    await render(
+      hbs`<Hds::Alert @type="inline" @title="This is the title" @description="This is the description" />`
+    );
+    assert.dom('.hds-alert__title').hasText('This is the title');
+    assert.dom('.hds-alert__description').hasText('This is the description');
+  });
+  test('it should render rich HTML when the @description argument contains HTML tags', async function (assert) {
+    assert.expect(8);
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="Hello <strong>strong</strong> and <em>em</em> and <code>code</code> and <a href='#'>link</a>" />`
+    );
+    assert.dom('.hds-alert__description strong').exists().hasText('strong');
+    assert.dom('.hds-alert__description em').exists().hasText('em');
+    assert.dom('.hds-alert__description code').exists().hasText('code');
+    assert.dom('.hds-alert__description a').exists().hasText('link');
+  });
+
+  // ACTIONS
+
+  test('it should render an Hds::Button component yielded to the "actions" container', async function (assert) {
+    assert.expect(2);
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="yo" id="test-alert" as |A|><A.Button @text="I am a button" @size="small" @color="secondary" /></Hds::Alert>`
+    );
+    assert
+      .dom(
+        '#test-alert .hds-alert__actions button.hds-button.hds-button--size-small.hds-button--color-secondary'
+      )
+      .exists()
+      .hasText('I am a button');
+  });
+  test('it should render an Hds::Link::Standalone component yielded to the "actions" container', async function (assert) {
+    assert.expect(2);
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="yo" id="test-alert" as |A|><A.Link::Standalone @icon="plus" @text="I am a link" href="#" @size="small" @color="secondary" /></Hds::Alert>`
+    );
+    assert
+      .dom(
+        '#test-alert .hds-alert__actions a.hds-link-standalone.hds-link-standalone--size-small.hds-link-standalone--color-secondary'
+      )
+      .exists()
+      .hasText('I am a link');
+  });
+
+  // GENERIC
+
+  test('it should render any content passed to the "generic" contextual component', async function (assert) {
+    assert.expect(2);
+    await render(
+      hbs`<Hds::Alert @type="inline" @description="yo" id="test-alert" as |A|><A.Generic><pre>test</pre></A.Generic></Hds::Alert>`
+    );
+    assert.dom('#test-alert .hds-alert__content pre').exists().hasText('test');
+  });
+
+  // DISMISS
+
+  test('it should not render the "dismiss" button by default', async function (assert) {
+    await render(hbs`<Hds::Alert @type="inline" @title="yo" />`);
+    assert.dom('button.hds-alert__dismiss').doesNotExist();
+  });
+  // TODO! how can this be done??
```

Comment:
what is the question here? how to create a fake callback function to pass into the component or something else?

Reply:
Yes, because this:
```
hbs`<Hds::Alert @description="yo" @onDismiss={{NOOP}} />`
``` is not a component invocation but a string so I am not sure how to "pass" a function to the handlebars (`hbs`) interpreter (in this case, the `NOOP` function (or any inline function).

If you comment the `NOOP` function at line 6, change `skip` to `test` and run the test, you will see this error:
<img width="715" alt="screenshot_1414" src="https://user-images.githubusercontent.com/686239/167144969-48d3d210-0f13-4d8c-8283-ae1e4c111650.png">

Reply:
I need to resolve this so I can try to make a suggestion in the UI. Wish me luck.

Reply:
Fixed in https://github.com/hashicorp/design-system/pull/245/commits/b3c9b29a5c500830a9d97dbbb3a598d2d490d125